cmake/BLTMacros.cmake

```diff
@@ -1425,3 +1425,74 @@ macro(blt_convert_to_system_includes)
     unset(_include_dirs)
 endmacro()
 
+
+##------------------------------------------------------------------------------
+## blt_check_code_compiles(CODE_COMPILES <variable>
+##                         VERBOSE_OUTPUT <ON|OFF (default OFF)>
+##                         SOURCE_STRING <quoted C++ program>)
+##
+## This macro checks if a snippet of C++ code compiles.
+##
+## SOURCE_STRING The source snippet to compile.
+## Must be a valid C++ program with a main() function.
+## Note: This parameter should be passed in as a quoted string variable. Otherwise,
+## cmake will convert the string into a list and lose the semicolons.
+## E.g. blt_check_code_compiles(SOURCE_STRING "${str_var}" ...)
+##
+## CODE_COMPILES A boolean variable the contains the compilation result.
+##
+## VERBOSE_OUTPUT Optional parameter to output debug information (Default: off)
+##------------------------------------------------------------------------------
+macro(blt_check_code_compiles)
+
+    set(options)
+    set(singleValueArgs CODE_COMPILES VERBOSE_OUTPUT)
+    set(multiValueArgs SOURCE_STRING)
```

Comment:
Does `SOURCE_STRING` need to be a `multiValueArg` because it can have embedded semicolons?

Reply:
i bet it shouldn't actually.

Reply:
Tried changing this to a single value arg and it removed all the semi-colons.